src/librustc/ty/context.rs

```diff
@@ -100,8 +100,8 @@ pub struct GlobalArenas<'tcx> {
     generics: TypedArena<ty::Generics>,
     trait_def: TypedArena<ty::TraitDef>,
     adt_def: TypedArena<ty::AdtDef>,
-    steal_mir: TypedArena<Steal<Mir<'tcx>>>,
-    mir: TypedArena<Mir<'tcx>>,
+    steal_mir: TypedArena<Steal<Body<'tcx>>>,
+    mir: TypedArena<Body<'tcx>>,
```

Comment:
I would avoid importing `Body` like this, especially outside of `rustc::mir`, `rustc_mir`, or `rustc_codegen_ssa`, and instead refer to it as `mir::Body`.
Also, the names of these arena fields should probably be `steal_mir_body` and `mir_body`.

Reply:
I'm gonna wait for @oli here.

Reply:
yea, this should be `mir::Body` so it's clear from a glance what's going on.

the field renames can happen in this PR, they aren't ever touched by other PRs, so there's no churn